CRM/Case/XMLProcessor/Report.php

```diff
@@ -956,26 +956,14 @@ public static function printCaseReport() {
 
     // Retrieve custom values for cases.
     $customValues = CRM_Core_BAO_CustomValueTable::getEntityValues($caseID, 'Case');
-    $extends = array('case');
-    $groupTree = CRM_Core_BAO_CustomGroup::getGroupDetail(NULL, NULL, $extends);
-    $caseCustomFields = array();
-    foreach ($groupTree as $gid => $group_values) {
-      foreach ($group_values['fields'] as $id => $field_values) {
-        if (array_key_exists($id, $customValues)) {
-          $caseCustomFields[$gid]['title'] = $group_values['title'];
-          $caseCustomFields[$gid]['values'][$id] = array(
-            'label' => $field_values['label'],
-            'value' => $customValues[$id],
-          );
-        }
-      }
-    }
+    $groupTree = CRM_Core_BAO_CustomGroup::getTree('Case', NULL, $caseID);
```

Comment:
Need to include the case type in the getTree() call. Otherwise it doesn't show custom fields that are specific to that case type. See e.g. https://github.com/civicrm/civicrm-core/blob/5.22.1/CRM/Case/Form/CaseView.php#L146